neo/rawio/cedrawio.py

```diff
@@ -0,0 +1,177 @@
+            stream_name = f'{stream_id} {num_chans}chans'
+            signal_streams.append((stream_name, stream_id))
+        signal_streams = np.array(signal_streams, dtype=_signal_stream_dtype)
+
+        # spike channels not handled
+        spike_channels = []
+        spike_channels = np.array([], dtype=_spike_channel_dtype)
+
+        # event channels not handled
+        event_channels = []
+        event_channels = np.array(event_channels, dtype=_event_channel_dtype)
+
+        self._seg_t_start = np.inf
+        self._seg_t_stop = -np.inf
+        for info in self.stream_info:
+            self._seg_t_start = min(self._seg_t_start,
+                                    info['first_time'] / info['sampling_rate'])
+            self._seg_t_stop = max(self._seg_t_stop,
+                                   info['max_time'] / info['sampling_rate'])
+
+        self.header = {}
+        self.header['nb_block'] = 1
+        self.header['nb_segment'] = [1]
+        self.header['signal_streams'] = signal_streams
+        self.header['signal_channels'] = signal_channels
+        self.header['spike_channels'] = spike_channels
+        self.header['event_channels'] = event_channels
+
+        self._generate_minimal_annotations()
+
+    def _segment_t_start(self, block_index, seg_index):
+        return self._seg_t_start
+
+    def _segment_t_stop(self, block_index, seg_index):
+        return self._seg_t_stop
+
+    def _get_signal_size(self, block_index, seg_index, stream_index):
+        size = self.stream_info[stream_index]['size']
+        return size
+
+    def _get_signal_t_start(self, block_index, seg_index, stream_index):
+        info = self.stream_info[stream_index]
+        t_start = info['first_time'] / info['sampling_rate']
+        return t_start
+
+    def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop,
+                                stream_index, channel_indexes):
+
+        if i_start is None:
+            i_start = 0
+        if i_stop is None:
+            i_stop = self.stream_info[stream_index]['size']
+
+        stream_id = self.header['signal_streams']['id'][stream_index]
+        signal_channels = self.header['signal_channels']
+        mask = signal_channels['stream_id'] == stream_id
+        signal_channels = signal_channels[mask]
+        if channel_indexes is not None:
+            signal_channels = signal_channels[channel_indexes]
+
+        num_chans = len(signal_channels)
+
+        size = i_stop - i_start
+        sigs = np.zeros((size, num_chans), dtype='int16')
+
+        info = self.stream_info[stream_index]
+        tFrom = info['first_time'] + info['divide'] * i_start
+        tUpto = info['first_time'] + info['divide'] * i_stop
+
+        for i, chan_id in enumerate(signal_channels['id']):
+            chan_ind = int(chan_id)
+            sig = self.smrx_file.ReadInts(chan=chan_ind,
+                    nMax=size, tFrom=tFrom, tUpto=tUpto)
+            sigs[:, i] = sig
```

Comment:
I think we should keep our variable names style consistent across Neo.
```suggestion
        t_from = info['first_time'] + info['divide'] * i_start
        t_upto = info['first_time'] + info['divide'] * i_stop

        for i, chan_id in enumerate(signal_channels['id']):
            chan_ind = int(chan_id)
            sig = self.smrx_file.ReadInts(chan=chan_ind,
                    nMax=size, tFrom=t_from, tUpto=t_upto)
            sigs[:, i] = sig
```